module/VuFind/src/VuFind/Search/Solr/Options.php

```diff
@@ -160,42 +160,27 @@ public function __construct(\VuFind\Config\PluginManager $configLoader)
         }
 
         // Load facet preferences
-        $facetSettings = $configLoader->get($this->facetsIni);
-        if (
-            isset($facetSettings->Advanced_Settings->translated_facets)
-            && count($facetSettings->Advanced_Settings->translated_facets) > 0
-        ) {
-            $this->setTranslatedFacets(
-                $facetSettings->Advanced_Settings->translated_facets->toArray()
-            );
+        if ($translatedFacets = $this->facetSettings['Advanced_Settings']['translated_facets'] ?? null) {
+            $this->setTranslatedFacets((array)$translatedFacets);
         }
-        if (isset($facetSettings->Advanced_Settings->delimiter)) {
-            $this->setDefaultFacetDelimiter(
-                $facetSettings->Advanced_Settings->delimiter
-            );
+        if ($delimiter = $this->facetSettings['Advanced_Settings']['delimiter'] ?? null) {
+            $this->setDefaultFacetDelimiter($delimiter);
         }
-        if (
-            isset($facetSettings->Advanced_Settings->delimited_facets)
-            && count($facetSettings->Advanced_Settings->delimited_facets) > 0
-        ) {
-            $this->setDelimitedFacets(
-                $facetSettings->Advanced_Settings->delimited_facets->toArray()
-            );
+        if ($delimitedFacets = $this->facetSettings['Advanced_Settings']['translated_facets'] ?? null) {
```

Comment:
Looks like a copy-and-paste error here; I think this should be:

```suggestion
        if ($delimitedFacets = $this->facetSettings['Advanced_Settings']['delimited_facets'] ?? null) {
```

Reply:
Oops, thanks for spotting that!